.travis.yml

```diff
@@ -15,38 +15,23 @@ addons:
             - libnl-route-3-200
             - libnl-route-3-dev
             - libnuma-dev
-# required for rdma-core
-            - build-essential
-            - debhelper
-            - dh-systemd
-            - fakeroot
-            - gcc
-            - git
-            - libudev-dev
-            - make
-            - ninja-build
-            - pandoc
-            - pkg-config
-            - python
-            - valgrind
-            - sparse
-            - wget
-            - abi-compliance-checker
-            - abi-dumper
-# 32 bit support packages
-            - gcc-multilib
-    ssh_known_hosts:
-        - www.openfabrics.org
-        - git.kernel.org
+            - rdma-core
+            - libibverbs-dev
+            - librdmacm-dev
 
 env:
     global:
         - PREFIX=$HOME/install
         - PATH=$PREFIX/bin:$PATH
-        - CPPFLAGS="-Werror -I$PREFIX/include"
+        - CPPFLAGS="-I$PREFIX/include"
         - LDFLAGS=-L$PREFIX/lib
         - LD_LIBRARY_PATH=$PREFIX/lib
         - LIBFABRIC_CONFIGURE_ARGS="--prefix=$PREFIX --enable-sockets"
+        # Temporarily disable -Werror testing (Jan 2020) because
+        # there are some warnings about unaligned atomics that I
+        # do not know how to fix
+        #- MAKE_FLAGS="AM_CFLAGS=-Werror"
+        - MAKE_FLAGS=
 
 # Brew update GNU Autotools so that autogen can succeed
 before_install:
```

Comment:
If I may pile on to the list of updates given you are revamping the config :-), I think there's value in having a pre-build code lint step here to catch some obvious issues before getting into the build tests. With EFA PRs, we _try_ to run checkpatch from the Linux kernel, given libfabric coding style [resembles that of the Linux kernel](https://github.com/ofiwg/libfabric/wiki#how-to-contribute).

Something like this maybe (might have to tune that to work with the different branches)?

```
wget https://raw.githubusercontent.com/torvalds/linux/master/scripts/checkpatch.pl
git diff master | ./checkpatch.pl --ignore LINUX_VERSION_CODE,GERRIT_CHANGE_ID,CONST_STRUCT,FILE_PATH_CHANGES,SPDX_LICENSE_TAG,COMMIT_MESSAGE,CAMELCASE,PREFER_KERNEL_TYPES,BIT_MACRO --show-types --no-tree --max-line-length=120 --strict --showfile
```

Alternatively, we can also add a separate Github Action just for linting. Either of these can be done outside this PR (I'd be happy to add that after your changes get merged too).

Reply:
I don't have much of an opinion here, either.  FWIW, I know a bunch of the usnic provider does _not_ adhere to the Linux kernel guidelines, but that was because it was written long before the Libfabric community expressed an opinion on the matter.

I think it would probably be slightly nicer for the style check to be a different CI line item.  That way, you still get all the correctness checking separately (i.e., red X or green check), and will have a separate item to indicate "you failed style checking".  If you could do that in a separate PR, I think that would be awesome.

Reply:
> I think it would probably be slightly nicer for the style check to be a different CI line item. 

Yep, I think we can achieve this with a Github action. Will set something up.